File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/GsonUtil.java
package com.elveselimoski.starterapp.json;

import android.net.Uri;
import androidx.annotation.VisibleForTesting;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.Date;

public final class GsonUtil {

    @VisibleForTesting
    private static GsonBuilder defaultGsonBuilder = new GsonBuilder()
            .registerTypeHierarchyAdapter(Uri.class, new UriTypeAdapter().nullSafe())
            .registerTypeHierarchyAdapter(Date.class, new DateTypeAdapter().nullSafe());
    public static Gson defaultGson = defaultGsonBuilder.create();

    private GsonUtil() {
    }
}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/GsonUtil.kt
package com.elveselimoski.starterapp.json

import android.net.Uri
import androidx.annotation.VisibleForTesting
import com.google.gson.Gson
import com.google.gson.GsonBuilder

import java.util.Date

object GsonUtil {

    @get:VisibleForTesting
    val defaultGsonBuilder: GsonBuilder = GsonBuilder()
        .registerTypeHierarchyAdapter(Uri::class.java, UriTypeAdapter().nullSafe())
        .registerTypeHierarchyAdapter(Date::class.java, DateTypeAdapter().nullSafe())

    val defaultGson: Gson = defaultGsonBuilder.create()
}
